integration_tests/testprovider/TestProvider.csproj

```
@@ -2,7 +2,8 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```

Similar to my other comment above.

Note: I was planning to update the fallback value for when the envvar isn't set from `net6.0` to `net8.0` in a follow-up change.

```suggestion
    <TargetFramework Condition=" '$(TARGET_FRAMEWORK)' != '' ">$(TARGET_FRAMEWORK)</TargetFramework>
    <TargetFramework Condition=" '$(TARGET_FRAMEWORK)' == '' ">net6.0</TargetFramework>
```